crates/shared/src/price_estimation/native/oneinch.rs

```diff
@@ -0,0 +1,140 @@
+use {
+    super::{NativePriceEstimateResult, NativePriceEstimating},
+    crate::price_estimation::PriceEstimationError,
+    anyhow::{anyhow, Result},
+    ethrpc::current_block::{into_stream, CurrentBlockStream},
+    futures::{future::BoxFuture, FutureExt, StreamExt},
+    primitive_types::{H160, U256},
+    reqwest::Client,
+    std::{
+        collections::HashMap,
+        sync::{Arc, Mutex},
+    },
+};
+
+const BASE_URL: &str = "https://api.1inch.dev/";
+
+struct OneInch {
+    // Denominated in wei
+    prices: Arc<Mutex<HashMap<H160, U256>>>,
+}
+
+impl OneInch {
+    #[allow(dead_code)]
+    pub fn new(client: Client, chain_id: u64, current_block: CurrentBlockStream) -> Self {
+        let instance = Self {
+            prices: Arc::new(Mutex::new(HashMap::new())),
+        };
+        instance.update_prices_in_background(client, chain_id, current_block);
+        instance
+    }
+
+    fn update_prices_in_background(
+        &self,
+        client: Client,
+        chain_id: u64,
+        current_block: CurrentBlockStream,
+    ) {
+        let prices = self.prices.clone();
+        tokio::task::spawn(async move {
+            let mut block_stream = into_stream(current_block);
+            loop {
+                match update_prices(&client, chain_id).await {
+                    Ok(new_prices) => {
+                        tracing::debug!("OneInch spot prices updated");
```

Comment:
We probably don't need this log, not even under `trace` level.

Reply:
I think it's helpful to know when a backrun job completes (in order to see for instance if we ever update the cache or how outdated the cache was for a specific quote).